runtime/array_decl.inl

```diff
@@ -45,25 +45,39 @@ enum class overwrite_element {
   NO
 };
 
+using list_entry_pointer_type = uint32_t;
+
+struct array_list_hash_entry {
+  list_entry_pointer_type next;
+  list_entry_pointer_type prev;
+};
+
+struct array_inner_control {
+  // This stub field is needed to ensure the alignment during the const arrays generation
+  // TODO: figure out something smarter
+  int stub;
+  int ref_cnt;
+  int64_t max_key;
+  array_list_hash_entry last;
+  uint32_t int_size;
+  uint32_t int_buf_size;
+  uint32_t string_size;
+  uint32_t string_buf_size;
+};
+
 template<class T>
 class array {
-
 public:
+  // TODO why we need 2 value types?
```

Comment:
`ValueType` is used in `TlDependentTypesUsings::deduce_params_from_type_tree()` 
`value_type` seems to be unused